ktor-http/common/src/io/ktor/http/URLBuilder.kt

```diff
@@ -240,27 +209,19 @@ public val URLBuilder.authority: String
         }
     }
 
-/**
- * Adds [components] to current [encodedPath]
- */
-public fun URLBuilder.pathComponents(components: List<String>): URLBuilder {
-    var paths = components
-        .map { part -> part.dropWhile { it == '/' }.dropLastWhile { it == '/' }.encodeURLQueryComponent() }
-        .filter { it.isNotEmpty() }
-        .joinToString("/")
-
-    // make sure that there's a slash separator at the end of current path
-    if (!encodedPath.endsWith('/')) {
-        paths = "/$paths"
+public var URLBuilder.encodedPath: String
+    get() {
+        val path = encodedPathSegments.joinToString("/")
+        return if (encodedPathSegments.isEmpty() || path.startsWith('/')) path else "/$path"
+    }
+    set(value) {
+        encodedPathSegments = when (value) {
+            "" -> mutableListOf()
+            "/" -> mutableListOf("")
+            else -> value.split('/').toMutableList().apply {
+                if (size > 1 && first().isEmpty()) {
+                    removeAt(0)
+                }
+            }
+        }
     }
-    encodedPath += paths
-
-    return this
-}
-
-/**
- * Adds [components] to current [encodedPath]
- */
-public fun URLBuilder.pathComponents(vararg components: String): URLBuilder {
```

Comment:
Could you tell me why we removing this method and do we have the migration?

Reply:
Added it back with deprecation in favor of new method